primary/public-src/ts/script-editform.ts

```diff
@@ -0,0 +1,229 @@
+import type { AbsoluteValueOperation, CompareOperation, DivideOperation, LogOperation, MinMaxOperation, MultiplyOperation, MultiselectOperation, operand, SumOperation } from 'scoutradioz-types';
+
+type LayoutEdit = import('scoutradioz-types').LayoutEdit;
+type StringDict = import('scoutradioz-types').StringDict;
+type DerivedLayout = import('scoutradioz-types').DerivedLayout;
+
+// consolesdflkjd
+async function validate() {
+	// get the JSON from the form
+	const jsonFieldElem = document.getElementById('jsonfield') as HTMLInputElement;
+	let jsonText = jsonFieldElem.value;
+
+	const previousKeysElem = document.getElementById('previousKeys') as HTMLInputElement;
+	let previousKeys = JSON.parse(previousKeysElem.value);
+	lightAssert(Array.isArray(previousKeys), 'previousKeys is not an array');
+	//console.log("previousKeys.length=" + previousKeys.length);
+
+	const formTypeElem = document.getElementById('form_type') as HTMLInputElement;
+	let formType = formTypeElem.value;
+	console.log('formType=' + formType);
+
+	// parse to JSON (or fail out with an error)
+	let jsonData: LayoutEdit[] = [];
+	try {
+		jsonData = JSON.parse(jsonText);
+	}
+	catch (error) {
+		if (error instanceof SyntaxError) {
+			NotificationCard.error(`Invalid JSON! Please correct. \n${error}`, { ttl: 0, exitable: true });
+		}
+		else {
+			NotificationCard.error('Invalid JSON! Please correct.');
+		}
+		return null;
+	}
+
+	// make sure 'order' and 'id' are unique; also clear out extraneous stuff
+	let idDict: StringDict = {};
```

Comment:
If we only care about the keys, Set is probably [better](https://github.com/anvaka/set-vs-object) than an object